src/service.rs

```diff
@@ -399,6 +402,21 @@ where
             .send_to(bus, source, ServiceId::Wallet, request)
             .map_err(From::from)
     }
+
+    fn send_client_ctl(
+        &mut self,
+        bus: ServiceBus,
+        senders: &mut Endpoints,
+        dest: ServiceId,
+        request: Request,
+    ) -> Result<(), Error> {
+        if let ServiceId::GrpcdClient(_) = dest {
+            senders.send_to(bus, dest, ServiceId::Grpcd, request)?;
+        } else {
+            senders.send_to(bus, self.identity(), dest, request)?;
+        }
+        Ok(())
+    }
```

Comment:
`service.rs` is definitely not a place that should be generic. It contains common definitions for the microservices. This is also the original author's vision. Look for example to [service.rs in lnp-node](https://github.com/LNP-WG/lnp-node/blob/master/src/service.rs#L214). It's `send_rpc` function only reports back to a single client, very similar to what I have defined here. The whole point of having this service function is that the microservices don't have to care about these internals when issuing the request. Otherwise they would have to populate a unique id and carry it through all requests every time we send something back to the client.